examples/chip-tool/config/PersistentStorage.cpp

```diff
@@ -204,3 +208,44 @@ LogCategory PersistentStorage::GetLoggingLevel()
 
     return chipLogLevel;
 }
+
+NodeId PersistentStorage::GetNodeId(const char * key, NodeId defaultVal)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    uint64_t nodeId;
+    uint16_t size = static_cast<uint16_t>(sizeof(nodeId));
+    err           = SyncGetKeyValue(key, &nodeId, size);
+    if (err == CHIP_NO_ERROR)
+    {
+        return static_cast<NodeId>(Encoding::LittleEndian::HostSwap64(nodeId));
+    }
+
+    return defaultVal;
+}
+
+NodeId PersistentStorage::GetLocalNodeId()
+{
+    return GetNodeId(kLocalNodeIdKey, kTestControllerNodeId);
+}
+
+NodeId PersistentStorage::GetRemoteNodeId()
+{
+    return GetNodeId(kRemoteNodeIdKey, kTestDeviceNodeId);
+}
+
+CHIP_ERROR PersistentStorage::SetNodeId(const char * key, NodeId value)
+{
+    uint64_t nodeId = Encoding::LittleEndian::HostSwap64(value);
```

Comment:
Why the swap here? And why wouldn't that be managed by `SyncSetKeyValue`. The storage will always be read back by the same platform that stored it.

Reply:
> Why the swap here?

General good practice from being burned too often in the past.

> And why wouldn't that be managed by SyncSetKeyValue

Because SyncSetKeyValue is passed a `void*`, so it can't do any swapping?  It's a pure byte blob store.

> The storage will always be read back by the same platform that stored it.

That is probably true in our immediate use case here; I've just been burned too many times before by writing persistent data on one platform and reading it on another one because network filesystems or people moving data between computers or whatnot....

I could remove the swap if you'd prefer.